tests/unit/model/state-test.js

```diff
@@ -6,138 +6,193 @@ import { isArray } from '@ember/array';
 
 import DefaultSchema from 'ember-m3/services/m3-schema';
 
-module('unit/model/state', function(hooks) {
-  setupTest(hooks);
-
-  hooks.beforeEach(function() {
-    this.store = this.owner.lookup('service:store');
-
-    class TestSchema extends DefaultSchema {
-      includesModel() {
-        return true;
-      }
-
-      computeAttributeReference(key, value, modelName, schemaInterface) {
-        let refValue = schemaInterface.getAttr(`*${key}`);
-        if (typeof refValue === 'string') {
-          return {
-            type: null,
-            id: refValue,
-          };
-        } else if (Array.isArray(refValue)) {
-          return refValue.map(x => ({
-            type: null,
-            id: x,
-          }));
-        }
-        return null;
-      }
+let computeNestedModel = function computeNestedModel(key, value) {
+  if (value && typeof value === 'object' && value.constructor !== Date && !isArray(value)) {
```

Comment:
I believe we can clean up this check by removing the Date check (seems no date usage below).

Reply:
Cleaned up